std/random.d

```diff
@@ -121,9 +124,41 @@ import std.traits;
     auto r = uniform(0.0L, 100.0L, rnd);
     assert(r >= 0 && r < 100);
 
+    // Sample from a custom type
+    enum Fruit { apple, mango, pear }
+    auto f = rnd.uniform!Fruit;
+    with(Fruit)
+    assert(f.among(apple, mango, pear));
+
     // Generate a 32-bit random number
     auto u = uniform!uint(rnd);
     static assert(is(typeof(u) == uint));
+
+    // Generate a random number in the range in the range [0, 1)
+    auto u2 = uniform01(rnd);
+    assert(u2 >= 0 && u2 < 1);
+
+    // Select an element randomly
+    auto el = 10.iota.choice(rnd);
+    assert(0 <= el && el < 10);
+
+    // Throw a dice with custom proportions
+    // 0: 20%, 1: 10%, 2: 60%
+    auto val = rnd.dice(0.2, 0.1, 0.6);
+    assert(0 <= val && val <= 2);
+
+    auto rnd2 = MinstdRand0(42);
+
+    // Select a random subsample from a range
+    assert(10.iota.randomSample(3, rnd2).equal([7, 8, 9]));
+
+    // Cover all elements in an array in random order
+    version(X86_64) // FIXME
+    assert(10.iota.randomCover(rnd2).equal([7, 4, 2, 0, 1, 6, 8, 3, 9, 5]));
+
+    // Shuffle an array
+    version(X86_64) // FIXME
```

Comment:
I would replace all instances of `FIXME` with `Issue XXXXXX`

Reply:
Sure. Done.